File: crates/gen-guest-rust/tests/variants.rs
#[allow(unused_imports, unused_variables, dead_code)]
#[rustfmt::skip]
#[allow(clippy::all)]
pub mod variants {
    use ::tauri_bindgen_guest_rust::serde;
    use ::tauri_bindgen_guest_rust::bitflags;
    #[derive(serde::Serialize, serde::Deserialize)]
    #[derive(Debug, Clone, PartialEq)]
    pub enum E1 {
        A,
    }
    #[derive(serde::Serialize, serde::Deserialize)]
    #[derive(Debug, Clone, PartialEq)]
    pub enum U1 {
        U32(u32),
        F32(f32),
    }
    #[derive(serde::Serialize, serde::Deserialize)]
    #[derive(Debug, Clone, PartialEq)]
    pub struct Empty {}
    #[derive(serde::Serialize)]
    #[derive(Debug, Clone, PartialEq)]
    pub enum V1Param<'a> {
        A,
        B(U1),
        C(E1),
        D(&'a str),
        E(Empty),
        F,
        G(u32),
    }
    #[derive(serde::Deserialize)]
    #[derive(Debug, Clone, PartialEq)]
    pub enum V1Result {
        A,
        B(U1),
        C(E1),
        D(String),
        E(Empty),
        F,
        G(u32),
    }
    #[derive(serde::Serialize, serde::Deserialize)]
    #[derive(Debug, Clone, PartialEq)]
    pub enum Casts1 {
        A(i32),
        B(f32),
    }
    #[derive(serde::Serialize, serde::Deserialize)]
    #[derive(Debug, Clone, PartialEq)]
    pub enum Casts2 {
        A(f64),
        B(f32),
    }
    #[derive(serde::Serialize, serde::Deserialize)]
    #[derive(Debug, Clone, PartialEq)]
    pub enum Casts3 {
        A(f64),
        B(u64),
    }
    #[derive(serde::Serialize, serde::Deserialize)]
    #[derive(Debug, Clone, PartialEq)]
    pub enum Casts4 {
        A(u32),
        B(i64),
    }
    #[derive(serde::Serialize, serde::Deserialize)]
    #[derive(Debug, Clone, PartialEq)]
    pub enum Casts5 {
        A(f32),
        B(i64),
    }
    #[derive(serde::Serialize, serde::Deserialize)]
    #[derive(Debug, Clone, PartialEq)]
    pub enum Casts6 {
        A((f32, u32)),
        B((u32, u32)),
    }
    #[derive(serde::Deserialize)]
    #[derive(Debug, Clone, PartialEq)]
    pub enum MyErrno {
        Bad1,
        Bad2,
    }
    #[derive(serde::Serialize)]
    #[derive(Debug, Clone, PartialEq)]
    pub struct IsCloneParam<'a> {
        #[serde(borrow)]
        pub v1: V1Param<'a>,
    }
    #[derive(serde::Deserialize)]
    #[derive(Debug, Clone, PartialEq)]
    pub struct IsCloneResult {
        pub v1: V1Result,
    }
    pub async fn e1_arg(x: E1) {
        ::tauri_bindgen_guest_rust::invoke("variants", "e1_arg", &(x)).await.unwrap()
    }
    pub async fn e1_result() -> E1 {
        ::tauri_bindgen_guest_rust::invoke("variants", "e1_result", &()).await.unwrap()
    }
    pub async fn u1_arg(x: U1) {
        ::tauri_bindgen_guest_rust::invoke("variants", "u1_arg", &(x)).await.unwrap()
    }
    pub async fn u1_result() -> U1 {
        ::tauri_bindgen_guest_rust::invoke("variants", "u1_result", &()).await.unwrap()
    }
    pub async fn v1_arg(x: V1Param<'_>) {
        ::tauri_bindgen_guest_rust::invoke("variants", "v1_arg", &(x)).await.unwrap()
    }
    pub async fn v1_result() -> V1Result {
        ::tauri_bindgen_guest_rust::invoke("variants", "v1_result", &()).await.unwrap()
    }
    pub async fn bool_arg(x: bool) {
        ::tauri_bindgen_guest_rust::invoke("variants", "bool_arg", &(x)).await.unwrap()
    }
    pub async fn bool_result() -> bool {
        ::tauri_bindgen_guest_rust::invoke("variants", "bool_result", &()).await.unwrap()
    }
    pub async fn option_arg(
        a: Option<bool>,
        b: Option<()>,
        c: Option<u32>,
        d: Option<E1>,
        e: Option<f32>,
        f: Option<U1>,
        g: Option<Option<bool>>,
    ) {
        ::tauri_bindgen_guest_rust::invoke(
                "variants",
                "option_arg",
                &(a, b, c, d, e, f, g),
            )
            .await
            .unwrap()
    }
    pub async fn option_result() -> (
        Option<bool>,
        Option<()>,
        Option<u32>,
        Option<E1>,
        Option<f32>,
        Option<U1>,
        Option<Option<bool>>,
    ) {
        ::tauri_bindgen_guest_rust::invoke("variants", "option_result", &())
            .await
            .unwrap()
    }
    pub async fn casts(
        a: Casts1,
        b: Casts2,
        c: Casts3,
        d: Casts4,
        e: Casts5,
        f: Casts6,
    ) -> (Casts1, Casts2, Casts3, Casts4, Casts5, Casts6) {
        ::tauri_bindgen_guest_rust::invoke("variants", "casts", &(a, b, c, d, e, f))
            .await
            .unwrap()
    }
    pub async fn result_arg(
        a: Result<(), ()>,
        b: Result<(), E1>,
        c: Result<E1, ()>,
        d: Result<(), ()>,
        e: Result<u32, V1Param<'_>>,
        f: Result<&'_ str, &'_ [u8]>,
    ) {
        ::tauri_bindgen_guest_rust::invoke("variants", "result_arg", &(a, b, c, d, e, f))
            .await
            .unwrap()
    }
    pub async fn result_result() -> (
        Result<(), ()>,
        Result<(), E1>,
        Result<E1, ()>,
        Result<(), ()>,
        Result<u32, V1Result>,
        Result<String, Vec<u8>>,
    ) {
        ::tauri_bindgen_guest_rust::invoke("variants", "result_result", &())
            .await
            .unwrap()
    }
    pub async fn return_result_sugar() -> Result<i32, MyErrno> {
        ::tauri_bindgen_guest_rust::invoke("variants", "return_result_sugar", &())
            .await
            .unwrap()
    }
    pub async fn return_result_sugar2() -> Result<(), MyErrno> {
        ::tauri_bindgen_guest_rust::invoke("variants", "return_result_sugar2", &())
            .await
            .unwrap()
    }
    pub async fn return_result_sugar3() -> Result<MyErrno, MyErrno> {
        ::tauri_bindgen_guest_rust::invoke("variants", "return_result_sugar3", &())
            .await
            .unwrap()
    }
    pub async fn return_result_sugar4() -> Result<(i32, u32), MyErrno> {
        ::tauri_bindgen_guest_rust::invoke("variants", "return_result_sugar4", &())
            .await
            .unwrap()
    }
    pub async fn return_option_sugar() -> Option<i32> {
        ::tauri_bindgen_guest_rust::invoke("variants", "return_option_sugar", &())
            .await
            .unwrap()
    }
    pub async fn return_option_sugar2() -> Option<MyErrno> {
        ::tauri_bindgen_guest_rust::invoke("variants", "return_option_sugar2", &())
            .await
            .unwrap()
    }
    pub async fn result_simple() -> Result<u32, i32> {
        ::tauri_bindgen_guest_rust::invoke("variants", "result_simple", &())
            .await
            .unwrap()
    }
    pub async fn is_clone_arg(a: IsCloneParam<'_>) {
        ::tauri_bindgen_guest_rust::invoke("variants", "is_clone_arg", &(a))
            .await
            .unwrap()
    }
    pub async fn is_clone_return() -> IsCloneResult {
        ::tauri_bindgen_guest_rust::invoke("variants", "is_clone_return", &())
            .await
            .unwrap()
    }
    pub async fn return_named_option() -> Option<u8> {
        ::tauri_bindgen_guest_rust::invoke("variants", "return_named_option", &())
            .await
            .unwrap()
    }
    pub async fn return_named_result() -> Result<u8, MyErrno> {
        ::tauri_bindgen_guest_rust::invoke("variants", "return_named_result", &())
            .await
            .unwrap()
    }
}

File: crates/gen-guest-dart/tests/variants.ts
export type Result<T, E> = { tag: 'ok', val: T } | { tag: 'err', val: E };


export enum E1 { 
A,
 }

export type U1 = 
number
 | 
number
;

export interface Empty {  }

export interface V1A { tag: 0 }

export interface V1B { tag: 1, value: U1 }

export interface V1C { tag: 2, value: E1 }

export interface V1D { tag: 3, value: string }

export interface V1E { tag: 4, value: Empty }

export interface V1F { tag: 5 }

export interface V1G { tag: 6, value: number }


export type V1 = 
V1A | 
V1B | 
V1C | 
V1D | 
V1E | 
V1F | 
V1G

export interface Casts1A { tag: 0, value: number }

export interface Casts1B { tag: 1, value: number }


export type Casts1 = 
Casts1A | 
Casts1B

export interface Casts2A { tag: 0, value: number }

export interface Casts2B { tag: 1, value: number }


export type Casts2 = 
Casts2A | 
Casts2B

export interface Casts3A { tag: 0, value: number }

export interface Casts3B { tag: 1, value: bigint }


export type Casts3 = 
Casts3A | 
Casts3B

export interface Casts4A { tag: 0, value: number }

export interface Casts4B { tag: 1, value: bigint }


export type Casts4 = 
Casts4A | 
Casts4B

export interface Casts5A { tag: 0, value: number }

export interface Casts5B { tag: 1, value: bigint }


export type Casts5 = 
Casts5A | 
Casts5B

export interface Casts6A { tag: 0, value: [number, number] }

export interface Casts6B { tag: 1, value: [number, number] }


export type Casts6 = 
Casts6A | 
Casts6B

export enum MyErrno { 
Bad1,

Bad2,
 }

export interface IsClone { 
v1: V1,
 }


            
            export async function e1Arg (x: E1)  {
            }
        
            
            export async function e1Result () : Promise<E1> {
            }
        
            
            export async function u1Arg (x: U1)  {
            }
        
            
            export async function u1Result () : Promise<U1> {
            }
        
            
            export async function v1Arg (x: V1)  {
            }
        
            
            export async function v1Result () : Promise<V1> {
            }
        
            
            export async function boolArg (x: boolean)  {
            }
        
            
            export async function boolResult () : Promise<boolean> {
            }
        
            
            export async function optionArg (a: boolean | null, b: [] | null, c: number | null, d: E1 | null, e: number | null, f: U1 | null, g: boolean | null | null)  {
            }
        
            
            export async function optionResult () : Promise<[boolean | null, [] | null, number | null, E1 | null, number | null, U1 | null, boolean | null | null]> {
            }
        
            
            export async function casts (a: Casts1, b: Casts2, c: Casts3, d: Casts4, e: Casts5, f: Casts6) : Promise<[Casts1, Casts2, Casts3, Casts4, Casts5, Casts6]> {
            }
        
            
            export async function resultArg (a: Result<_, _>, b: Result<_, E1>, c: Result<E1, _>, d: Result<[], []>, e: Result<number, V1>, f: Result<string, Uint8Array[]>)  {
            }
        
            
            export async function resultResult () : Promise<[Result<_, _>, Result<_, E1>, Result<E1, _>, Result<[], []>, Result<number, V1>, Result<string, Uint8Array[]>]> {
            }
        
            
            export async function returnResultSugar () : Promise<Result<number, MyErrno>> {
            }
        
            
            export async function returnResultSugar2 () : Promise<Result<_, MyErrno>> {
            }
        
            
            export async function returnResultSugar3 () : Promise<Result<MyErrno, MyErrno>> {
            }
        
            
            export async function returnResultSugar4 () : Promise<Result<[number, number], MyErrno>> {
            }
        
            
            export async function returnOptionSugar () : Promise<number | null> {
            }
        
            
            export async function returnOptionSugar2 () : Promise<MyErrno | null> {
            }
        
            
            export async function resultSimple () : Promise<Result<number, number>> {
            }
        
            
            export async function isCloneArg (a: IsClone)  {
            }
        
            
            export async function isCloneReturn () : Promise<IsClone> {
            }
        
            
            export async function returnNamedOption () : Promise<number | null> {
            }
        
            
            export async function returnNamedResult () : Promise<Result<number, MyErrno>> {
            }
        